libcodechecker/server/api/store_handler.py

```diff
@@ -290,81 +291,35 @@ def setRunDuration(storage_session, run_id, duration):
-        checker_cat = main_section['category']
-        bug_type = main_section['type']
-        line_num = main_section['location']['line']
-        column = main_section['location']['col']
+        checker_name = main_section['check_name']
+        context = generic_package_context.get_context()
+        severity_name = context.severity_map.get(checker_name,
+                                                 'UNSPECIFIED')
+        severity = ttypes.Severity._NAMES_TO_VALUES[severity_name]
 
-        checker_id = checker_id or 'NOT FOUND'
-
-        # Check for duplicates by bug hash.
-        LOG.debug("checking duplicates")
-        for report in all_reports:
-            LOG.debug("there is a possible duplicate")
-            # TODO: file_id and path equality check won't be necessary
-            # when path hash is added.
-            if report.checker_id == checker_id and \
-                    is_same_event_path(report.id, events, session):
-
-                new_status = None
-
-                if report.detection_status == 'new' and not \
-                        storage_session.is_touched(run_id, report.id) or \
-                        report.detection_status == 'unresolved' or \
-                        report.detection_status == 'reopened':
-                    new_status = 'unresolved'
-
-                    report.file_id = file_id
-                    report.line = line_num
-                    report.column = column
-
-                    report_path_and_events[report.id] = (bugpath, events)
-                elif report.detection_status == 'resolved':
-                    new_status = 'reopened'
-
-                    report.file_id = file_id
-                    report.line = line_num
-                    report.column = column
-                    report.fixed_at = None
-
-                    report_path_and_events[report.id] = (bugpath, events)
-
-                if new_status:
-                    report.detection_status = new_status
-
-                storage_session.touch_report(run_id, report.id)
-
-                return report.id
-
-        LOG.debug("no duplicate storing report")
         report = Report(run_id,
-                        bug_hash,
+                        main_section['issue_hash_content_of_line_in_context'],
                         file_id,
-                        msg,
-                        checker_id,
-                        checker_cat,
-                        bug_type,
-                        line_num,
-                        column,
+                        main_section['description'],
+                        checker_name or 'NOT FOUND',
+                        main_section['category'],
+                        main_section['type'],
+                        main_section['location']['line'],
+                        main_section['location']['col'],
                         severity,
-                        "new",
+                        detection_status,
```

Comment:
if the detection_status is unresolved, set the detection_date to the date of the original reports. This way, we will correctly preserve in the detection history the date when the reports with the bug hash was introduced.

Reply:
We should test that the detection dates are the same for all of the reports with the same bughash, and it should be the first detection date.